packages/components/psammead-radio-schedule/src/ProgramCard/index.jsx

```diff
@@ -0,0 +1,250 @@
+
+const SummaryWrapper = styled.p`
+  ${({ service }) => service && getSansRegular(service)};
+  ${({ script }) => script && getLongPrimer(script)};
+  color: ${C_METAL};
+  padding-bottom: ${GEL_SPACING_DBL};
+  margin: 0; /* Reset */
+`;
+
+const ButtonWrapper = styled.div`
+  ${({ service }) => service && getSansRegular(service)};
+  ${({ script }) => script && getMinion(script)};
+  padding: ${GEL_SPACING};
+  background-color: ${({ backgroundColor }) => backgroundColor};
+  color: ${({ durationColor }) => durationColor};
+  border-top: 1px solid transparent;
+`;
+
+const IconWrapper = styled.span.attrs({ 'aria-hidden': 'true' })`
+  > svg {
+    color: ${({ durationColor }) => durationColor};
+    fill: currentColor;
+    width: 1.0625rem;
+    height: ${GEL_SPACING_DBL};
+    margin: 0;
+  }
+`;
+
+const DurationWrapper = styled.span`
+  ${({ dir }) =>
+    dir === 'ltr'
+      ? `padding-left: ${GEL_SPACING};`
+      : `padding-right: ${GEL_SPACING};`}
+`;
+
+const programStateConfig = {
+  live: {
+    backgroundColor: C_POSTBOX,
+    headerLabelColor: C_POSTBOX,
+    headerTextColor: C_EBON,
+    titleColor: C_SHADOW,
+    durationColor: C_WHITE,
+  },
+  next: {
+    backgroundColor: C_WHITE,
+    headerLabelColor: C_KINGFISHER,
+    headerTextColor: C_METAL,
+    titleColor: C_METAL,
+    durationColor: C_KINGFISHER,
+  },
+  onDemand: {
+    backgroundColor: C_EBON,
+    headerTextColor: C_EBON,
+    titleColor: C_SHADOW,
+    durationColor: C_WHITE,
+  },
+};
+
+const renderHeaderContent = ({
+  type,
+  link,
+  translation,
+  brandTitle,
+  episodeTitle,
+  service,
+  script,
+  startTime,
+}) => {
+  const isOnDemand = type === 'onDemand';
+  const hiddenTextProps =
+    translation.toLowerCase() === 'live' ? { lang: 'en-GB' } : {};
+
+  const content = (
+    <>
+      {!isOnDemand && (
+        <VisuallyHiddenText {...hiddenTextProps}>
+          {`${translation}, `}
+        </VisuallyHiddenText>
+      )}
+      <VisuallyHiddenText>
+        {`${brandTitle}, ${startTime}, ${episodeTitle}`}
+      </VisuallyHiddenText>
+      {!isOnDemand && (
+        <LabelWrapper
+          service={service}
+          script={script}
+          {...programStateConfig[type]}
+        >
+          {`${translation.toUpperCase()} `}
+        </LabelWrapper>
+      )}
+      {brandTitle}
+      <TitleWrapper
+        service={service}
+        script={script}
+        {...programStateConfig[type]}
+      >
+        {episodeTitle}
+      </TitleWrapper>
+    </>
+  );
+
+  return type === 'next' ? content : <Link href={link}>{content}</Link>;
+};
+
+const ProgramCard = ({
+  dir,
+  service,
+  script,
+  brandTitle,
+  summary,
+  episodeTitle,
+  duration: { durationValue, durationText },
+  startTime,
+  state: { type, translation },
+  link,
+}) => (
+  <CardWrapper>
+    <TextWrapper>
+      <HeadingWrapper
+        service={service}
+        script={script}
+        {...programStateConfig[type]}
+      >
+        {renderHeaderContent({
```

Comment:
What do people think about exporting the Heading and composing it into the card component as a child or a prop? It's nice to avoid 'prop drilling' (https://kentcdodds.com/blog/prop-drilling), but it looks like here the rendering of the header has just been separated into a function to make this component more readable, so may be overkill for now.

Reply:
You're right, the separation here was just to make the code more readable and to apply the `link` conditionally.